frame/staking/src/lib.rs

```diff
@@ -1254,9 +1259,22 @@ decl_module! {
 			);
 
 			let ledger = ledger.rebond(value);
-
 			Self::update_ledger(&controller, &ledger);
 		}
+
+		/// Remove all data structure concerning a staker/stash once its balance is zero.
+		/// This is essentially equivalent to `withdraw_unbonded` except it can be called by anyone
+		/// and the target `stash` must have no funds left.
+		///
+		/// This can be called from any origin.
+		///
+		/// - `stash`: The stash account to reap. Its balance must be zero.
+		fn reap_stash(_origin, stash: T::AccountId) {
```

Comment:
How is this not an attack vector for a spam/ddos attack?

Reply:
how could it be?

Reply:
What stops anyone from calling this function as many times as they want, not incur a fee, and force the blockchain to do some storage reads and other calculations?

Reply:
it's not a free transaction for anyone other than the block producer (it is for them, but if they want to waste their block, they'll incur an according opportunity cost).